.pmd/ruleset.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<ruleset name="DoD"
+         description="pragmatic approach to maintaining code quality in large-scale"
```

Comment:
strange that its not detected:

is this:
- supposed to be like this?
- any good?

@adangel 

- https://github.com/pmd/pmd/issues/1991
- https://github.com/pmd/pmd/issues/470
- https://github.com/pmd/pmd/blob/main/pmd-core/src/main/resources/ruleset_2_0_0.xsd#L11

<img width="1070" alt="image" src="https://github.com/user-attachments/assets/62acaa97-9c92-463e-a211-896811fba0bc" />


<img width="747" alt="image" src="https://github.com/user-attachments/assets/83b0a202-e727-475c-80d1-e1b62ccde77f" />

Reply:
> strange that its not detected:
> 
> is this:
> 
>     * supposed to be like this?
> 
>     * any good?

Assuming you are talking about your "problem", that the ruleset description is not detected and PMD says it is missing. Is this correct? Please be more clear in what you write/ask/..., thanks! I always need to guess what you are talking about...

The ruleset description must be a _element_, and not an attribute, so:

```xml
<ruleset>
  <description>My description</description>
...
</ruleset>
```

See also https://docs.pmd-code.org/latest/pmd_userdocs_making_rulesets.html

Reply:
ah thanks